### PR TITLE
persist: support interchange of ShardId via its string format

### PR DESCRIPTION
**Pull request by danhhz**
TBD whether this is necessary for production, but it's convenient for
demos/examples/etc.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

### CODE REVIEW

**Comment by danhhz** [comment]
Closing in favor of #11757

File: src/persist-client/src/error.rs [hunk 1]
```diff
@@ -23,6 +23,14 @@ impl std::fmt::Display for InvalidUsage {
 
 impl std::error::Error for InvalidUsage {}
 
+#[cfg(test)]
+impl PartialEq for InvalidUsage {
+    fn eq(&self, other: &Self) -> bool {
+        // TODO: What are the right semantics here?
```
Comment (aljoscha) [inline src/persist-client/src/error.rs:29]: Seems fine for `cfg(test)`

**Review by aljoscha — APPROVED** [review]
lgtm!

File: src/persist-client/src/error.rs (hunk 1 above)
Comment (danhhz) [reply src/persist-client/src/error.rs:29]: yeah, I think I agree. done. also removed the one this was copied from (ExternalError)

**Review by danhhz — COMMENTED** [review]
TFTR!

**Pull request closed without merging**